docs/basic/transactions.rst

```diff
@@ -297,6 +298,67 @@ but not entirely committed yet.
     # and the program would continue from here with no exception.
 
 
+.. _transaction-status:
+
+Transaction status
+^^^^^^^^^^^^^^^^^^
+
+Each `Transaction` object exposes a `~Transaction.status` property that allows
+you to inspect the current state of the transaction. This can be useful for
+debugging, logging, or implementing custom transaction management logic.
+
+The `!status` property returns a `Transaction.Status` enum value that can be
+one of the following:
+
+- `!NOT_STARTED`: The transaction has been created but not yet entered (before
+  the `!__enter__()` method is called).
+- `!ACTIVE`: The transaction is currently active (inside the `!with` block).
+- `!COMMITTED`: The transaction exited successfully and changes were committed.
+- `!ROLLED_BACK_EXPLICITLY`: The transaction was explicitly rolled back, either
+  by raising a `Rollback` exception or by using `!force_rollback=True`.
+- `!ROLLED_BACK_WITH_ERROR`: The transaction was rolled back due to an
+  exception raised within the transaction block.
+- `!FAILED`: The transaction failed due to a connection failure (e.g., the
+  connection was closed or became unavailable during the transaction).
+
+Here's an example of how to use the status property:
+
+.. code:: python
+
+    from psycopg import Rollback
+
+    # Successful transaction
+    with conn.transaction() as tx:
+        assert tx.status.name == "ACTIVE"
+        conn.execute("INSERT INTO data VALUES (%s)", ("Hello",))
+    assert tx.status.name == "COMMITTED"
```

Comment:
Wouldn't it be better to suggest to access the enums by identity rather than by name?

```suggestion
    assert tx.status.name == tx.Status.COMMITTED
```